sdk/eventhub/Microsoft.Azure.EventHubs/src/EventDataBatch.cs

```diff
@@ -80,16 +82,27 @@ public bool TryAdd(EventData eventData)
 
         long GetEventSizeForBatch(EventData eventData)
         {
-            // Create AMQP message here. We will use the same message while sending to save compute time.
+            // Create AMQP message here.
```

Comment:
Approach looks solid to me.  This seems like it would benefit from some test coverage for:

- If you `TryAdd` the same messages with and without diagnostics enabled, do you see the overhead being taken into account and fewer messages being accepted?
- If you saturate a batch to its maximum size with diagnostics enabled, does the overhead allow you to exceed the maximum size?
- (Live) If you attempt to send a maximum size batch with and without diagnostics enabled, does the service reject the one with diagnostics enabled?

Reply:
Good points Jesse. Added 2 tests to cover your scenarios.